packages/material-ui/src/BottomNavigation/BottomNavigation.test.js

```diff
@@ -41,81 +42,86 @@ describe('<BottomNavigation />', () => {
   );
 
   it('renders with a null child', () => {
-    const wrapper = shallow(
+    const { container } = render(
       <BottomNavigation showLabels value={0}>
         <BottomNavigationAction label="One" />
         {null}
         <BottomNavigationAction label="Three" />
       </BottomNavigation>,
     );
-    assert.strictEqual(wrapper.find(BottomNavigationAction).length, 2);
+    expect(getBottomNavigation(container).childNodes.length).to.equal(2);
   });
 
   it('should pass selected prop to children', () => {
-    const wrapper = shallow(
+    const { container } = render(
       <BottomNavigation showLabels value={1}>
         <BottomNavigationAction icon={icon} />
         <BottomNavigationAction icon={icon} />
       </BottomNavigation>,
     );
-    assert.strictEqual(wrapper.childAt(0).props().selected, false);
-    assert.strictEqual(wrapper.childAt(1).props().selected, true);
+    expect(getBottomNavigation(container).childNodes[0]).to.not.have.class('Mui-selected');
+    expect(getBottomNavigation(container).childNodes[1]).to.have.class('Mui-selected');
   });
 
   it('should overwrite parent showLabel prop', () => {
-    const wrapper = shallow(
+    const bottomNavigationProps = {};
+    const ControlledBottomNavigationAction = (props) => {
+      bottomNavigationProps[props['data-testid']] = props;
+      return <BottomNavigationAction {...props} />;
+    };
+    render(
       <BottomNavigation showLabels value={1}>
-        <BottomNavigationAction icon={icon} />
-        <BottomNavigationAction icon={icon} showLabel={false} />
+        <ControlledBottomNavigationAction icon={icon} data-testid="withLabel" />
+        <ControlledBottomNavigationAction
+          icon={icon}
+          showLabel={false}
+          data-testid="withoutLabel"
+        />
       </BottomNavigation>,
     );
-    assert.strictEqual(wrapper.childAt(0).props().showLabel, true);
-    assert.strictEqual(wrapper.childAt(1).props().showLabel, false);
+    expect(bottomNavigationProps.withLabel.showLabel).to.equal(true);
+    expect(bottomNavigationProps.withoutLabel.showLabel).to.equal(false);
```

Comment:
I have only a doubt here because I didn't find a better way to check props of child components

Reply:
Yeah, I think that it should be done the other way around. We should test the DOM implication not how props spread, it's an implementation detail.

Reply:
Yes I see that in a lot of components props are checked.
In this case there is no implication in DOM, test should be changed

Reply:
right, we should test the `iconOnly` class name

Reply:
yes I will do